### ISSUE TITLE
Put tests behind feature flag in CMakeLists.txt

### ISSUE THREAD

**Issue by studoot** [sharkdp/dbg-macro#42]
I'm using dbg-macro in my project by using the CMake `add_subdirectory` command. This brings in not just the library, but also the test executables. They don't get built, but the contained tests get added to CTest's list of tests to run, which means that CTest fails.

Looking at another [header-only library](https://github.com/TartanLlama/expected) that I use, I see that [it uses a feature flag to exclude the tests](https://github.com/TartanLlama/expected/blob/3d741708b967b83ca1e2888239196c4a67f9f9b0/CMakeLists.txt#L27) when not wanted. This could be done with dbg-macro by wrapping lines 10-19 in:

```cmake
if (DBG_MACRO_ENABLE_TESTS)
    :         :        :
(existing lines 10-19 here)
    :         :        :
endif()
```

I'm quite happy to create a pull-request for it...

**Comment by sharkdp** [comment]
> I'm quite happy to create a pull-request for it...

Sounds great, thank you for reporting this!

**Comment by sharkdp** [comment]
closed via #43
